### PR TITLE
Updated to Mapbox Android SDK 5.2.1

### PR DESCRIPTION
**Pull request by nitaliano**
* Updates to Mapbox Android SDK 5.2.1

### CODE REVIEW

File: docs/MapView.md [hunk 1]
```diff
@@ -23,6 +23,7 @@
 | attributionEnabled | `bool` | `true` | `false` | Enable/Disable attribution on map. For iOS you need to add MGLMapboxMetricsEnabledSettingShownInApp=YES<br/>to your Info.plist |
 | logoEnabled | `bool` | `true` | `false` | Enable/Disable the logo on the map. |
 | compassEnabled | `bool` | `none` | `false` | Enable/Disable the compass from appearing on the map |
+| textureMode | `bool` | `false` | `false` | Enable/Disable TextureMode insted of SurfaceView |
```
Comment (henrikra) [inline docs/MapView.md:26]: Typo `insted `

File: docs/docs.json [hunk 2]
```diff
@@ -1292,6 +1292,13 @@
         "default": "none",
         "description": "Enable/Disable the compass from appearing on the map"
       },
+      {
+        "name": "textureMode",
+        "required": false,
+        "type": "bool",
+        "default": "false",
+        "description": "Enable/Disable TextureMode insted of SurfaceView"
```
Comment (henrikra) [inline docs/docs.json:1300]: Typo `insted `

**Review by henrikra — COMMENTED** [review]
Great! Check these :D

**Comment by nitaliano** [comment]
It actually appears like 5.2.1 is breaking being able to create an offline region on Android

**Comment by tobrun** [comment]
This will be fixed with 5.2.2, aimed to be released next week. For now you can a workaround this by calling `FileSource.getInstance().activate()`.

**Comment by nitaliano** [comment]
Thanks @tobrun! I'll give that a go.

**Comment by nitaliano** [comment]
@butchmarshall This PR should fix the issue you are seeing https://github.com/mapbox/react-native-mapbox-gl/pull/889. @tobrun already fixed the issue in gl-native so that PR I just submitted will keep master in a working state until we update to 5.2.2

**Comment by butchmarshall** [comment]
Just updated to latest master (which includes #7eea08fc7cc54032920585cff9f6176681349d7d - this was the PR you were referencing right?).

Still seems to have the same issue.

**Comment by tobrun** [comment]
Validated upstream in gl-native, the offline region creation is working there with the 5.2.2 codebase.

**Comment by nitaliano** [comment]
thanks for posting updates back here @tobrun much appreciated!

@butchmarshall I noticed a black map today as well in the logs I noticed it gl-native had a warning for no internet connection. Maybe try closing out Android studio and your emulators and trying again? When I saw it happen today I killed adb and restarted the emulator and it started working fine again.

**Comment by butchmarshall** [comment]
Hmm - I don't actually use android studio.  I'll try and replicate with a fresh app.

**Pull request merged**